### ISSUE TITLE
Is it as expected that Polars has slower performance for simple arithmetic operations than Numpy? 

### ISSUE THREAD

**Issue by paladin158** [pola-rs/polars#8436]
### Research

- [X] I have searched the above polars tags on Stack Overflow for similar questions.

- [ ] I have asked my usage related question on [Stack Overflow](https://stackoverflow.com).

### Question about Polars

The task I am benchmarking with is simply **clipping values element-wise**. I have done this using both ```numpy``` and ```polars```. But, it turned out that using ```numpy``` is much faster (~5x) than using ```polars``` (as shown below).

So, my question:
- Is this behavior as expected?
- If so, does it mean that ```polars``` (although it is highly optimized for join/groupby), may not be well-suited for performing relatively simple numerical vector/array operations such as clipping in my example?

```python
import timeit
import numpy as np
import polars as pl

N = 10_000_000
x = np.random.normal(size=N)
y = np.random.normal(size=N)
z = y + 0.5
df = pl.DataFrame({"x": x, "y": y, "z": z})

>>> timeit.timeit(lambda: np.minimum(np.maximum(x, y), z), number=10)
0.60923

>>> timeit.timeit(lambda: df.select(pl.min(pl.max(pl.col("x"), pl.col("y")), pl.col("z"))), number=10)
3.39337
```

**Comment by s-banach** [comment]
I think numpy has fancy SIMD versions of more of these basic functions.
For example, I ran your benchmark with the latest numpy vs with numpy 1.20, before some of the latest SIMD updates.
- In latest numpy, polars takes 6x as long to compute maximum.
- In 1.20, polars takes 1.5 as long to compute maximum.

I think polars has added SIMD for some operations but not all.

**Comment by cjackal** [comment]
An optimization on horizontal min/max(#8463) has been merged recently, wonder if performance would be comparable to numpy after that.

**Comment by marcosmoova** [comment]
I just have just run it with polars version 0.17.8 and now are much closer.
`timeit.timeit(lambda: np.minimum(np.maximum(x, y), z), number=10)`
`0.28034829199896194`
`timeit.timeit(lambda: df.select(pl.min(pl.max(pl.col("x"), pl.col("y")), pl.col("z"))), number=10)`
`0.318892124996637`

**Comment by alexander-beedie** [comment]
Yup, that update was added following this Issue being raised; I think we can close this out now?
